Sail/src/Sail/entities/systems/Gameplay/candles/CandleHealthSystem.cpp

```diff
@@ -108,6 +108,17 @@ void CandleHealthSystem::update(float dt) {
 		}
 #pragma endregion
 
+		// Flicker effect for the torches
+		static float clockLightModifier = 0;
+		clockLightModifier += dt * (rand() % 100 / 100.0f);
+
+		//						Sine wave function                    +        random variance
+		float r = (sinf(clockLightModifier * 50.0f) * 0.075f + 0.55f) + (rand() % 10 - 5) / 50.0f;
+		LightComponent* light = e->getComponent<LightComponent>();
+		light->defaultColor = glm::vec3(r + 0.1f, (r - 0.05f) * 0.5f, (r - 0.1f) * 0.25f);
+		// save this line for further possible further testing in future
+		//light->getPointLight().setAttenuation(0.0f, 0.0f, r );
+
```

Comment:
The LightComponent is being fetch twice. Here and in a line just below this part. Maybe use the same variable?